back/app/services/authentication_service.rb

```diff
@@ -33,4 +33,22 @@ def logout_url(provider, user)
   def supports_logout?(provider)
     method_by_provider(provider).respond_to? :logout_url
   end
+
+  def prevent_user_account_hijacking(user)
+    # When a user created an account with someone else's email, chose a password, but could not
+    # verify the email, we don't want that user to hijack the real email owner after they create
+    # an account (with SSO).
+    return nil if !user
+
+    if AppConfiguration.instance.feature_activated? 'user_confirmation'
+      if user.password_digest && user.confirmation_required?
```

Comment:
`user.confirmation_required?` already checked the AppConfiguration object.

I think we should also check that `email_confirmed_at` is nil as there is possibly a way where an existing user can choose email only confirmation for an action and reset their requirement to confirm - not entirely sure, but if the email has been confirmed then there is a chance we have data in the platform for this user

Reply:
Alright, thanks for spotting, I changed this part of the code.